src/zenml/integrations/mlflow/services/mlflow_deployment.py

```diff
@@ -185,9 +185,12 @@ def predict(self, request: "NDArray[Any]") -> "NDArray[Any]":
                 "Please start the service before making predictions."
             )
 
-        response = requests.post(
-            self.endpoint.prediction_uri,
-            json={"instances": request.tolist()},
-        )
+        if self.endpoint.prediction_uri is not None:
+            response = requests.post(
+                self.endpoint.prediction_uri,
+                json={"instances": request.tolist()},
+            )
+        else:
+            raise ValueError("No endpoint known for prediction.")
```

Comment:
How are these changes related to our service? Was mypy complaining?

Reply:
Indeed it was. `self.endpoint` is an `Optional`, so the `.prediction_uri` does not always exist